#690 custom anno 4

djencks opened the pull request:
Implements Peter's idea for an XMLAttribute meta-annotation.

bjhargrave commented on biz.aQute.bndlib/src/aQute/bnd/annotation/xml/XMLAttribute.java:
```diff
@@ -0,0 +1,54 @@
+
+	String prefix() default "";
+
+	String[] embedIn();
```
Could this default to "*"?

bjhargrave commented:
I don't really have much expertise here, but I don't have any objections to this.

djencks commented:
I improved the defaults processing per BJ's comment: not specifying the namespace now actually works and embedIn defaults to {"*"}

djencks commented:
Changed how classes are represented in the Annotation backing map to be TypeRefs and added a test that now Class valued annotation elements are processed correctly.
